beacon-chain/core/altair/epoch_precompute.go

```diff
@@ -61,6 +61,43 @@ func InitializeEpochValidators(ctx context.Context, state iface.BeaconStateAltai
 	return pValidators, bal, nil
 }
 
+// ProcessInactivityScores of beacon chain. This updates inactivity scores of beacon chain and
+// updates the precompute validator struct for later processing.
+func ProcessInactivityScores(
+	ctx context.Context,
+	state iface.BeaconState,
+	vals []*precompute.Validator,
+) (iface.BeaconState, []*precompute.Validator, error) {
+	inactivityScores, err := state.InactivityScores()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for i, v := range vals {
+		if v.IsPrevEpochTargetAttester && !v.IsSlashed {
+			// Decrease inactivity score when validator gets target correct.
+			if v.InactivityScore > 0 {
+				v.InactivityScore -= 1
+			}
+		} else {
+			prevEpoch := helpers.PrevEpoch(state)
+			finalizedEpoch := state.FinalizedCheckpointEpoch()
+			if helpers.IsInInactivityLeak(prevEpoch, finalizedEpoch) {
+				// Increase validator's inactivity score by bias when validator didn't vote target.
+				v.InactivityScore += params.BeaconConfig().InactivityScoreBias
```

Comment:
In the original code this line was executed when `!v.IsPrevEpochTargetAttester`. Currently it will be executed whenever `v.IsPrevEpochTargetAttester && !v.IsSlashed` as false, which means either when `!v.IsPrevEpochTargetAttester` (which is correct) or when `v.IsPrevEpochTargetAttester && v.IsSlashed` (which seems incorrect).

Reply:
This is correct and the old implementation is incorrect. Thanks for mentioning this, I will open a PR to fix the old spot